Realm/RLMObjectBase.mm

```diff
@@ -133,24 +106,48 @@ + (RLMObjectSchema *)sharedSchema {
     return RLMSchema.sharedSchema[self.className];
 }
 
-- (NSArray *)linkingObjectsOfClass:(NSString *)className forProperty:(NSString *)property {
-    if (!_realm) {
+void RLMObjectBaseSetRealm(__unsafe_unretained RLMObjectBase *object, __unsafe_unretained RLMRealm *realm) {
```

Comment:
I don't think we need these nil checks -- these functions appear to only ever be called on self or non-optional objects. Currently most of the C functions don't check for nil, and doing it inconsistently is kinda dangerous.